File: client/MdNotes mac/MdNotes/exec.swift
import Foundation
import Ink

struct Node: Codable, Hashable {
    let local_id: Int
    let global_id: Int
    var name: String
    var naiyou: String?
    
    var revison: Int
    var is_shared: Bool
    var dirty: Bool
    var is_deleted: Bool
    
    let created_at: Date
    var updated_at: Date
    
    func return_html() -> String {
        guard let md = naiyou, !md.isEmpty else { return "" }
        let parser = MarkdownParser()
        return parser.html(from: md)
    }

    init(local_id: Int,
         global_id: Int,
         name: String,
         naiyou: String? = nil,
         revison: Int = 0,
         is_shared: Bool = false,
         dirty: Bool = false,
         is_deleted: Bool = false,
         created_at: Date = Date(),
         updated_at: Date = Date()) {
        self.local_id = local_id
        self.global_id = global_id
        self.name = name
        self.naiyou = naiyou
        self.revison = revison
        self.is_shared = is_shared
        self.dirty = dirty
        self.is_deleted = is_deleted
        self.created_at = created_at
        self.updated_at = updated_at
    }
}

File: client/MdNotes pc/exec.cs
using System;
using Markdig;

public class Node
{
	public int LocalId { get; }
	public int GlobalId { get; }
	public string Name { get; set; }
	public string? Naiyou { get; set; }

	public int Revison { get; set; }
	public bool IsShared { get; set; }
	public bool Dirty { get; set; }
	public bool IsDeleted { get; set; }

	public DateTime CreatedAt { get; }
	public DateTime UpdatedAt { get; set; }

	public Node(int localId,
				int globalId,
				string name,
				string? naiyou = null,
				int revison = 0,
				bool isShared = false,
				bool dirty = false,
				bool isDeleted = false,
				DateTime? createdAt = null,
				DateTime? updatedAt = null)
	{
		LocalId = localId;
		GlobalId = globalId;
		Name = name;
		Naiyou = naiyou;
		Revison = revison;
		IsShared = isShared;
		Dirty = dirty;
		IsDeleted = isDeleted;
		CreatedAt = createdAt ?? DateTime.UtcNow;
		UpdatedAt = updatedAt ?? DateTime.UtcNow;
	}

	public string ReturnHtml()
	{
		if (string.IsNullOrEmpty(Naiyou)) return string.Empty;
		var pipeline = new MarkdownPipelineBuilder().UseAdvancedExtensions().Build();
		return Markdown.ToHtml(Naiyou, pipeline);
	}
}

